### ISSUE TITLE
Is it possible to write a unstructured cell whose connectivity is defined with points at other processor?

### ISSUE THREAD

**Issue by Blumenkranz** [JuliaVTK/WriteVTK.jl#157]

**Comment by jipolanco** [comment]
Hi, I'd need a bit more details and context to be able to answer. What do you mean by other processor? Are you using threads, distributed computing or something else?

**Comment by jipolanco** [comment]
WriteVTK  is not MPI-aware, which means that you need to explicitly exchange data between MPI processes if needed.

But I still don't understand what kind of strategy you want to use for writing files:

- Do you want a single MPI process to write a VTK file including all cells? In that case you need to explicitly send all data to that process before writing the file in serial. (This doesn't scale very well when you have many processes.)
- Or do you want each MPI process to write its local cells? In that case, have you looked at the [parallel file formats](https://juliavtk.github.io/WriteVTK.jl/dev/metadata/parallel/)? I haven't tried it myself, but I'd guess that a single MPI process would need to know all the points in a cell (including those belonging to neighbour processes), and then you would get repeated points as you mentioned. But maybe ParaView knows how to handle that? I'd say it's a very common usecase.

**Comment by Blumenkranz** [comment]
The strategy I want to adopt is the latter. I tried the parallel file format in WriteVTK, and it seems that they do not support ghost cells (I'm not entirely sure, but I tried my best). When I defined the complete list of points in each rank, the mesh separation occurred. Unfortunately, Paraview also encountered troubles here (we indeed did not inform Paraview that these duplicate points are the same points).
